orangecontrib/text/preprocess.py

```diff
@@ -1,94 +1,298 @@
+
+        Removes the stop words specified in the stop_words parameter.
+
+        Args:
+            data (list): The input document in the form of a list.
+            stop_words (set): The set of stop word we wish to use.
+
+        Returns:
+            list: Pre-processed document in the form of a list, holding the
+                tokens.
+        """
+        assert isinstance(stop_words, set)
+        return [word for word in data if word not in stop_words]
+
+    def _get_corpus_specific_stopwords(self, corpus_documents):
+        """Calculates the corpus specific stop words.
+
+        Constructs a stop word list, using the minimum and maximum document
+        frequency parameters.
+
+        Args:
+            corpus_documents (list): The input corpus structured as a list of
+                lists.
+
+        Returns:
+            list: Document frequency stop words in a list.
+        """
+        # 1.) First, we need to tokenize the documents in the corpus.
+        corpus_documents = [self._tokenize(doc) for doc in corpus_documents]
+
+        document_frequencies = {}
+        # 2.) Calculate the document frequency for individual tokens.
+        for document in corpus_documents:
+            for token in set(document):
+                document_frequencies[token] = document_frequencies.get(
+                        token, 0) + 1
+
+        # 3.) SW based on minimum document frequency.
+        sw_min = set()
+        if self.min_df is not None:
+            min_df = self.min_df  # If int.
+            if isinstance(self.min_df, float):  # If float.
+                min_df = round(len(corpus_documents) * self.min_df)
+            sw_min = {token for token, freq in document_frequencies.items()
+                      if freq <= min_df}
+
+        # 4.) SW based on maximum document frequency.
+        sw_max = set()
+        if self.max_df is not None:
+            max_df = self.max_df  # If int.
+            if isinstance(self.max_df, float):  # If float.
+                max_df = round(len(corpus_documents) * self.max_df)
+            sw_max = {token for token, freq in document_frequencies.items()
+                      if freq >= max_df}
+
+        self.df_stop_words = sw_min | sw_max
+
+    def _check_df_constraints(self, min_df, max_df):
+        """Checks for appropriate data types and parameter ranges.
+
+        Args:
+            min_df (int, float): The bottom threshold for document frequency
+                check.
+            max_df (int, float): The upper threshold for document frequency
+                check.
+
+        Raises:
+            ValueError: If any faulty inputs are detected, determined by type
+                and variable range.
+        """
+        if min_df is not None:
+            self.use_df_sw = True
+            is_float = isinstance(min_df, float)
+            is_int = isinstance(min_df, int)
+            if is_float:
+                in_range = 1.0 >= min_df >= 0.0
```

Comment:
I might be a bit picky, but change you change this to: `0.0 <= min_df <= 1.0`? Same at line 251.